source/templates.tex

```diff
@@ -6492,7 +6492,7 @@
 when determining whether the constraints are satisfied
 or as specified in \ref{temp.constr.decl} when comparing declarations.
 \begin{note}
-The satisfaction of constraints is determined during name lookup or overload
+The satisfaction of constraints is determined during overload
```

Comment:
I think technically the satisfaction of constraints *is* determined during name lookup for an *conversion-function-id*.

Reply:
I don't think I follow (it doesn't help that we don't really describe name lookup for a *conversion-function-id* beyond hints like [basic.lookup.classref]/7).  Are you saying that
```cpp
struct A {operator int();};
template<int=0> struct B : A {operator int() requires false;};
int f(B<> b) {return b;}
```
should work because name lookup discovers that `B<0>::operator int` is non-viable and proceeds to the base class?

Reply:
No, but
```
template<typename> struct A {
  template<typename T> requires (sizeof(T) == 1) operator T();
};
auto p = &A<int>::operator char;
```
... checks that the constraints of `A<int>::operator T<char>` are satisfied during name lookup.

Reply:
I'm sorry, I'm just still not getting it.  (Maybe P1787 has destroyed my ability to understand name lookup?)

My understanding there is that `A` being a template is irrelevant, that name lookup performs deduction and identifies `A<int>::operator T<char>` as the correct specialization to add to the (otherwise empty) overload set, and that overload resolution (of the [new kind](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2019/p1972r0.html) but without any contribution from [temp.deduct.funcaddr] in this case) _then_ checks the constraints ([over.over]/5) and either accepts or discards the one candidate (the latter, typically, in the situation where `A<int>::operator T<double>` was chosen by name lookup).

Reply:
Yeah, changing to "deduction and overload resolution" would be an improvement.

I think the difference between doing it during name lookup and deferring to overload resolution is probably only visible in a variant of your example such as

```
struct A { operator int(); };
template<bool B> struct B : A { template<same<int> T> operator T() requires B; };
```

... for which (unlike in your example) I think `B<false>::operator int` *would* select `A::operator int`.

Reply:
I'll merge this manually with that one change.

Reply:
Thanks for the fix (and the explanation).